### PR TITLE
Fix booking location

### PR DESCRIPTION
**Pull request by jerryfreshworks**
- Fix bug:  showing default "location value" when edit booking.

image

- Fix bug: limit 32 check disappeared abnormally.

- [sorting location](https://justice.gov.bc.ca/jira/browse/CSBISS-108):
image
image

- [refactor locationAutoComplete component](https://justice.gov.bc.ca/jira/browse/CSBISS-107)
image

### CODE REVIEW

File: api/src/distance/distance.service.ts [hunk 1]
```diff
@@ -101,7 +102,7 @@ export class DistanceService {
     );
 
     if (distanceLimit) {
-      intpWithDistance = intpWithDistance.filter(intp => intp.distance <= distanceLimit);
+      intpWithDistance = intpWithDistance.filter(intp => intp.distance <= DISTANCE_LIMIT);
```
Comment (michel-fresh) [inline api/src/distance/distance.service.ts:105]: interesting, so there is a table with distances for each interpreter to each location?

File: frontend/src/components/form/inputs/AutoCompleteField.tsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    <StyledFormLabel htmlFor={name}>{label}</StyledFormLabel>
+    <Field name={name}>
+      {({ field, form, ...props }: FieldProps) => (
+        <Autocomplete
+          options={options}
+          getOptionLabel={getOptionLabel}
+          id={name}
+          size="small"
+          onChange={onChange(form)}
+          // value={field.value}
```
Comment (michel-fresh) [inline frontend/src/components/form/inputs/AutoCompleteField.tsx:40]: maybe remove?

**Review by michel-fresh — APPROVED** [review]
cool, touches a lot of files but everything looks fine at first glance

File: api/src/distance/distance.service.ts (hunk 1 above)
Comment (jerryfreshworks) [reply api/src/distance/distance.service.ts:105]: ya, as we want to save the cost of using GoogleMapApi, so we have a "distance" table to save the existing interpreter and court house.

File: frontend/src/components/form/inputs/AutoCompleteField.tsx (hunk 2 above)
Comment (jerryfreshworks) [reply frontend/src/components/form/inputs/AutoCompleteField.tsx:40]: thanks for catching this.

**Review by fwpushan — COMMENTED** [review]
Great job

**Pull request merged**